simulated_input/src/sim.rs

```diff
@@ -169,14 +169,24 @@ fn main_impl() -> Result<()> {
                     Some((kind, val)) => match kind {
                         "tick" | "🕐" | "t" => {
                             let tick = str::parse::<u128>(val)?;
-                            #[cfg(feature = "simulated_output")]
-                            k.kbd_out.log.in_tick(tick);
+                            #[cfg(all(
+                                not(feature = "simulated_input"),
```

Comment:
Looking at this again, this repetition is quite messy, probably best to take the same approach as with `check_for_exit` - move this to a function in this file that takes the `kbd_out` by reference, with a conditionally-compiled body.

Reply:
That's party because autoformatting converts dense 1-liners into these 4-line mini-monsters!